File: .skills/openclaw-skills/skills/iswangheng/ai-dev-tools/mcp_server.py
#!/usr/bin/env python3
"""
MCP Server for SaaS Affiliate
让其他 Agent 可以通过 MCP 协议调用

使用方式：
1. 作为独立 MCP Server 运行
2. 或集成到其他支持 MCP 的 Agent 平台
"""

import json
import sys
from tools import (
    search_saas_tools,
    get_affiliate_link,
    get_all_products,
    get_product_details
)

def handle_request(method: str, params: dict = None):
    """处理 MCP 请求"""
    params = params or {}
    
    if method == "tools/list":
        return {
            "tools": [
                {
                    "name": "search_saas_tools",
                    "description": "根据用户需求搜索推荐 SaaS 工具",
                    "inputSchema": {
                        "type": "object",
                        "properties": {
                            "query": {"type": "string"}
                        },
                        "required": ["query"]
                    }
                },
                {
                    "name": "get_affiliate_link",
                    "description": "获取指定产品的 affiliate 链接",
                    "inputSchema": {
                        "type": "object",
                        "properties": {
                            "product_name": {"type": "string"}
                        },
                        "required": ["product_name"]
                    }
                },
                {
                    "name": "get_all_products",
                    "description": "获取所有可推荐的产品列表",
                    "inputSchema": {"type": "object", "properties": {}}
                },
                {
                    "name": "get_product_details",
                    "description": "获取产品详细信息",
                    "inputSchema": {
                        "type": "object",
                        "properties": {
                            "product_name": {"type": "string"}
                        },
                        "required": ["product_name"]
                    }
                }
            ]
        }
    
    elif method == "tools/call":
        tool_name = params.get("name")
        tool_args = params.get("arguments", {})
        
        if tool_name == "search_saas_tools":
            result = search_saas_tools(tool_args.get("query", ""))
        elif tool_name == "get_affiliate_link":
            result = get_affiliate_link(tool_args.get("product_name", ""))
        elif tool_name == "get_all_products":
            result = get_all_products()
        elif tool_name == "get_product_details":
            result = get_product_details(tool_args.get("product_name", ""))
        else:
            result = {"error": f"Unknown tool: {tool_name}"}
        
        return {"content": [{"type": "text", "text": json.dumps(result, ensure_ascii=False)}]}
    
    return {"error": f"Unknown method: {method}"}


# MCP Server 主循环
if __name__ == "__main__":
    # 读取 stdin，处理 MCP 协议
    while True:
        try:
            line = sys.stdin.readline()
            if not line:
                break
            
            request = json.loads(line.strip())
            response = handle_request(
                request.get("method", ""),
                request.get("params", {})
            )
            print(json.dumps(response), flush=True)
        except Exception as e:
            print(json.dumps({"error": str(e)}), flush=True)

File: .skills/openclaw-skills/skills/colaliang/yundian-woo-importer/mcp-server.ts
#!/usr/bin/env node
/**
 * Yundian+ WooCommerce Importer MCP Server
 * 
 * Usage in Cursor/Trae:
 * Command: node mcp-server.js
 * Env: YUNDIAN_WOO_IMPORTER_API_KEY=your_api_key, YUNDIAN_WOO_IMPORTER_API_URL=https://your-domain.com
 */

import { Server } from "@modelcontextprotocol/sdk/server/index.js";
import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
import {
  CallToolRequestSchema,
  ListToolsRequestSchema,
} from "@modelcontextprotocol/sdk/types.js";

const API_KEY = process.env.YUNDIAN_WOO_IMPORTER_API_KEY;
const API_URL = process.env.YUNDIAN_WOO_IMPORTER_API_URL || "http://localhost:3000"; // Note: Use the actual port where the app is running

if (!API_KEY) {
  console.error("Error: YUNDIAN_WOO_IMPORTER_API_KEY environment variable is required.");
  process.exit(1);
}

const server = new Server(
  {
    name: "yundian-woo-importer-mcp-server",
    version: "1.0.0",
  },
  {
    capabilities: {
      tools: {},
    },
  }
);

// Define tools
server.setRequestHandler(ListToolsRequestSchema, async () => {
  return {
    tools: [
      {
        name: "import_products",
        description: "Queue products for import from Shopify, Wix, WordPress, or Amazon to WooCommerce.",
        inputSchema: {
          type: "object",
          properties: {
            platform: {
              type: "string",
              enum: ["shopify", "wix", "wordpress", "amazon"],
              description: "The source platform"
            },
            shopifyBaseUrl: {
              type: "string",
              description: "Required for Shopify. E.g., https://example.myshopify.com"
            },
            wixUrl: {
              type: "string",
              description: "Required for Wix"
            },
            productLinks: {
              type: "array",
              items: { type: "string" },
              description: "List of product URLs or handles to import"
            },
            mode: {
              type: "string",
              enum: ["all"],
              description: "Set to 'all' to discover and import all products (Shopify only)"
            }
          },
          required: ["platform"]
        }
      },
      {
        name: "check_import_status",
        description: "Check the status of an import job using the requestId",
        inputSchema: {
          type: "object",
          properties: {
            requestId: {
              type: "string",
              description: "The request ID returned by import_products"
            }
          },
          required: ["requestId"]
        }
      }
    ]
  };
});

// Handle tool execution
server.setRequestHandler(CallToolRequestSchema, async (request) => {
  try {
    if (request.params.name === "import_products") {
      const response = await fetch(`${API_URL}/api/v1/import`, {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
          "Authorization": `Bearer ${API_KEY}`
        },
        body: JSON.stringify(request.params.arguments)
      });
      const data = await response.json();
      return {
        content: [{ type: "text", text: JSON.stringify(data, null, 2) }]
      };
    } 
    
    if (request.params.name === "check_import_status") {
      const { requestId } = request.params.arguments as any;
      const response = await fetch(`${API_URL}/api/v1/status?requestId=${requestId}`, {
        headers: {
          "Authorization": `Bearer ${API_KEY}`
        }
      });
      const data = await response.json();
      return {
        content: [{ type: "text", text: JSON.stringify(data, null, 2) }]
      };
    }

    throw new Error(`Unknown tool: ${request.params.name}`);
  } catch (error: any) {
    return {
      content: [{ type: "text", text: `Error: ${error.message}` }],
      isError: true
    };
  }
});

// Start the server
async function main() {
  const transport = new StdioServerTransport();
  await server.connect(transport);
  console.error("Yundian+ WooCommerce Importer MCP Server running on stdio");
}

main().catch((error) => {
  console.error("Server error:", error);
  process.exit(1);
});
